start-tests.sh

```diff
@@ -7,106 +7,267 @@
+# Check to see if OPTIONS_HUB_OC_IDP is unset or null.
+if [[ -z $OPTIONS_HUB_OC_IDP || "$OPTIONS_HUB_OC_IDP" == "null" ]]; then
+  log_color "purple" "OPTIONS_HUB_OC_IDP" "is (null or not set); setting to 'kube:admin'\n"
+  export OPTIONS_HUB_OC_IDP=kube:admin
+else
+  log_color "purple" "OPTIONS_HUB_OC_IDP" "detected, using $OPTIONS_HUB_OC_IDP for test.\n"
+fi
 
-# Check to see if CYPRESS_OC_IDP is null.
-if [[ -z $CYPRESS_OC_IDP || "$CYPRESS_OC_IDP" == "null" ]]; then
-  log_color "purple" "CYPRESS_OC_IDP" "is (null or not set); setting to 'kube:admin'\n"
-  export CYPRESS_OC_IDP=kube:admin
+# Create dir for kube configs.
+mkdir -p ./kube/config
+
+# Check to see if OPTIONS_HUB_BASEDOMAIN, OPTIONS_HUB_USER, or OPTIONS_HUB_PASSWORD are missing. We need these to run the UI test.
+if [[ -z $OPTIONS_HUB_BASEDOMAIN || -z $OPTIONS_HUB_USER || -z $OPTIONS_HUB_PASSWORD ]]; then
+  log_color "red" "One or more exported variables are undefined for hub cluster." "(set ${PURPLE}OPTIONS_HUB_BASEDOMAIN, OPTIONS_HUB_BASEDOMAIN, and OPTIONS_HUB_BASEDOMAIN${NC} to execute the test with environment variables)\n"
+
+  # Check to see if the kubeconfig for the hub cluster is available.
+  if [[ ! -f $OPTIONS_HUB_KUBECONFIG ]]; then
+    log_color "red" "The kubeconfig file for the hub cluster was not located." "(set ${PURPLE}KUBECONFIG${NC} to ${YELLOW}$OPTIONS_HUB_KUBECONFIG${NC} and oc login to create kubeconfig file)"
+    exit 1
+  else
+    # To run the E2E test, we need the password to log into the UI. If it's not exported or available from the options.yaml file, skip the UI test.
+    if [[ -z $OPTIONS_HUB_PASSWORD || "$OPTIONS_HUB_PASSWORD" == "null" ]]; then
+      log_color "purple" "OPTIONS_HUB_PASSWORD" "is required to login into the ACM console. ${YELLOW}Skipping UI test.${NC}\n"
+      export SKIP_UI_TEST=true
+    fi
+
+    echo -e "Kubeconfig file detected at: $OPTIONS_HUB_KUBECONFIG => copying to ./kube/config/hub-kubeconfig"
+    cp $OPTIONS_HUB_KUBECONFIG ./kube/config/hub-kubeconfig
+    export OPTIONS_HUB_KUBECONFIG=./kube/config/hub-kubeconfig
+
+    # Check to see if there are any kubecontext to be used from the hub cluster kubeconfig.
+    if [[ -z $OPTIONS_HUB_KUBECONTEXT || "$OPTIONS_HUB_KUBECONTEXT" == "null" ]]; then
+      HUB_CLUSTER=($(oc config get-clusters --kubeconfig=$OPTIONS_HUB_KUBECONFIG))
+      export OPTIONS_HUB_KUBECONTEXT=default/${HUB_CLUSTER[1]}/kube:admin
+    fi
+
+    echo -e
+    log_color "cyan" "Switching context to log into Kube API server"
+    oc config use-context --kubeconfig=$OPTIONS_HUB_KUBECONFIG $OPTIONS_HUB_KUBECONTEXT
+    
+    export OPTIONS_HUB_BASEDOMAIN=$(oc whoami --show-server=true | cut -d'.' -f2- | cut -d':' -f1)
+
+    if [[ $OPTIONS_HUB_OC_IDP == "kube:admin" && -z $OPTIONS_HUB_USER ]]; then
+      export OPTIONS_HUB_USER=kubeadmin
+    fi
+
+    log_color "purple" "HUB CLUSTER:" "$OPTIONS_HUB_BASEDOMAIN"
+  fi
```

Comment:
If the `kubeconfig` file is available for the `hub` cluster, we can switch the cluster context to the `hub's kubecontext`, only if the required environment variables aren't exported.